app/controllers/mixins/actions/provider_actions/mass_transform.rb

```diff
@@ -0,0 +1,56 @@
+module Mixins
+  module Actions
+    module ProviderActions
+      module MassTransform
+        def vm_transform_mass
+          dialog = Dialog.find_by!(:label => 'Transform VM')
+          @right_cell_text = _("Transform VMs to RHV")
+          if params.key?(:id)
+            provider = ExtManagementSystem.find_by!(:id => params[:id].to_i)
+            dialog_initialize(
+              dialog.resource_actions.first,
+              :header     => @right_cell_text,
+              :target_id  => provider.id,
+              :target_kls => Provider.name
+            )
+          else
+            provider = ExtManagementSystem.find_by!(:type => 'ManageIQ::Providers::Redhat::InfraManager')
```

Comment:
This bit surprises me, this is equivalent to `provider = ManageIQ::Providers::Redhat::InfraManager.first`, right? Meaning "I don't know which RHV this is, so just pick a random one".

Shouldn't we always know the provider?
Or do we really not care about the instance, only the type?

Reply:
.. never mind, already answered in https://github.com/ManageIQ/manageiq-ui-classic/pull/3073#discussion_r157895370